### PR TITLE
[teleop] Update GUI camera functionality

### PR DESCRIPTION
**Pull request by ErikLytle**
- Changed Cameras.lcm from "cam1 to cam8" to "cam0 to cam7"
- Separated microscope camera feed toggle from main cameras & dual stream
- Created "No Camera" button to deselect the current camera
- Fixed controls bug that would allow for more than 2 cameras to be selected
Screen Shot 2022-03-12 at 5 04 14 PM

### CODE REVIEW

File: base_station/gui/src/components/CameraSelection.vue [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <template>
   <div class="wrap2">
-    <span v-if="cam_index >= 0" class="title">Current Camera: {{cam_index}}</span>
+    <span v-if="cam_index >= 0" class="title">Current Camera: {{cam_index - 1}}</span>
     <div class="buttons">
       <template v-for="i in 3">
         <button class="cam_buttons" ref="cams" v-on:click="$emit('cam_index', i-1)"> <span>{{cameras[i-1]}}</span> </button>
```
Comment (tabiosg) [inline base_station/gui/src/components/CameraSelection.vue:6]: change to i - 2 so these cameras have indices -1, 0, and 1. do similar things for the rest.

File: base_station/gui/src/components/CameraSelection.vue [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <template>
   <div class="wrap2">
-    <span v-if="cam_index >= 0" class="title">Current Camera: {{cam_index}}</span>
+    <span v-if="cam_index >= 0" class="title">Current Camera: {{cam_index - 1}}</span>
```
Comment (tabiosg) [inline base_station/gui/src/components/CameraSelection.vue:3]: change to cam_index. (instead redefine what cam_index is by subtracting it in its definition

File: base_station/gui/src/components/CameraSelection.vue [hunk 3]
```diff
@@ -77,7 +77,7 @@
       sendSettings: function() {
         this.$parent.$parent.publish("/cam_settings", {
           type: 'CamSettings',
-          cam_index: this.cam_index,
+          cam_index: this.cam_index - 1,
```
Comment (tabiosg) [inline base_station/gui/src/components/CameraSelection.vue:80]: based on my comments from before, change back to cam_index

File: base_station/gui/src/components/Cameras.vue [hunk 4]
```diff
@@ -195,37 +205,45 @@
           if (new_index !== 0) {
             this.cams[this.cam_index_1 - 1] = false
```
Comment (tabiosg) [inline base_station/gui/src/components/Cameras.vue:206]: based on previous comments, change from cam_index_1 - 1 to just cam_index_1 and so on

**Review by tabiosg — CHANGES_REQUESTED** [review]
I'm only looking at the screenshot attached. In the screenshot, the camera buttons are from Camera 0 - Camera 7 while the Camera statuses on the top say Camera 1 - Camera 8. I suggest just keeping everything consistent and make everything 0 to 7. Also I had some comments to make the code more readable (like there are a lot of "camera_X - 1" instead of "camera_X". Other than that, the gui looks good!

**Comment by ErikLytle** [comment]
I just went through and updated it, the camera statuses should be good now and almost all of the indexing should match the shown camera value (the v-for loops use 1-indexing and the button disabling uses 0-indexing so they'll have to have - 1 etc).

**Pull request merged**